lib/util.h

```diff
@@ -121,7 +122,24 @@ void __msp_safe_free(void **ptr);
 
 #define msp_safe_free(pointer) __msp_safe_free((void **) &(pointer))
 
-size_t msp_binary_interval_search(double query, const double *values, size_t n_values);
+size_t idx_1st_upper_bound(const double *values, size_t n_values, double query);
+size_t idx_1st_strict_upper_bound(
+    const double *elements, size_t n_elements, double query);
+
+inline const double *
```

Comment:
Looks like these aren't being used at all in this PR - easiest to remove for now?

Reply:
Well, not *easiest* ... easiest would be leave 'em in. :) `pr/fast_search_lookup` is calling `ptr_1st_strict_upper_bound`.

But this does relate to a slightly annoying issue that is in these lines. No code, not even `pr/fast_search_lookup`, is currently calling `ptr_1st_upper_bound`. And I didn't bother writing any unit test to call it.

So I've been wonder whether I should:
A) leave it in and not worry about it,
B) add a unit test case to call `ptr_1st_upper_bound` for code coverage or,
C) remove `ptr_1st_upper_bound`

I've been proceeding along A) but let me know if you prefer B) or C). And by C) I mean remove not just for this PR #1223 but all PRs, forever, until some "real" code is actually going to call it.

Reply:
Well as a general rule I think we'd want tests for any code that's introduced in a PR. So, seems easiest to delete these functions for this PR?

Reply:
I'll go ahead and move them from this PR and into `pr/fast_search_lookup`. Although the not worrying about it option would be the easiest option.